### DISCUSSION TITLE
get current stage in pytorch-lightning

### DISCUSSION THREAD

**Discussion by 2533245542** [Lightning-AI/pytorch-lightning#12988] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi everyone,

I have a question related to pytorch lightning's module.

Say I am in the forward function and I want to see if I am in which of the train/val/test stage. 

Is there a way to do so?

Didn't get lucky with google search and a look into the source code seems like self.training is boolean for indicating training/non-training stage, but still not enough.

**Comment by rohitgr7** [comment]
use `self.trainer.training/testing/validating`
